front/lib/actions/mcp_internal_actions/servers/trigger_management.ts

```diff
@@ -0,0 +1,539 @@
+        ),
+    },
+    withToolLogging(
+      auth,
+      {
+        toolNameForMonitoring: "trigger_management_create_schedule",
+        agentLoopContext,
+      },
+      async ({ name, schedule, prompt, timezone }) => {
+        const owner = auth.workspace();
+        const user = auth.user();
+
+        if (!owner) {
+          return new Err(new MCPError("Workspace not found"));
+        }
+
+        if (!user) {
+          return new Err(new MCPError("User not found"));
+        }
+
+        if (!agentLoopContext?.runContext) {
+          return new Err(new MCPError("Agent context is required"));
+        }
+
+        const { agentConfiguration } = agentLoopContext.runContext;
+
+        const remaining = await rateLimiter({
+          key: `trigger_create:${owner.sId}:${user.sId}`,
+          maxPerTimeframe: 20,
+          timeframeSeconds: 86400,
+          logger,
+        });
+
+        if (remaining === 0) {
+          getStatsDClient().increment(
+            "tools.trigger_management.rate_limit_hit",
+            1,
+            [`workspace_id:${owner.sId}`, `agent_id:${agentConfiguration.sId}`]
+          );
+          return new Err(
+            new MCPError(
+              "Rate limit exceeded: You can create up to 20 triggers per day."
+            )
+          );
+        }
+
+        // Determine timezone: explicit param > context > default
+        const resolvedTimezone = timezone ?? getUserTimezone(agentLoopContext);
+
+        if (!resolvedTimezone) {
+          return new Err(new MCPError("Provide a timezone"));
+        }
+
+        const cronResult = await generateCronRule(auth, {
+          naturalDescription: schedule,
+          defaultTimezone: resolvedTimezone,
+        });
+
+        if (cronResult.isErr()) {
+          logger.error(
+            {
+              error: cronResult.error,
+              workspaceId: owner.id,
+              schedule,
+            },
+            "Error parsing schedule"
+          );
+          return new Err(
+            new MCPError(
+              `Unable to understand the schedule "${schedule}". Please try rephrasing (e.g., "every weekday at 9am", "every Monday at 10am").`
+            )
+          );
+        }
+
+        const { cron, timezone: resultTimezone } = cronResult.value;
+
+        const result = await TriggerResource.makeNew(auth, {
+          workspaceId: owner.id,
+          agentConfigurationId: agentConfiguration.sId,
+          name,
+          kind: "schedule",
+          enabled: true,
+          configuration: {
+            cron,
+            timezone: resultTimezone,
+          },
+          naturalLanguageDescription: schedule,
+          customPrompt: prompt,
+          editor: user.id,
+          webhookSourceViewId: null,
+          executionPerDayLimitOverride: null,
+          executionMode: null,
```

Comment:
shouldn't be null i think, should be "fair_use" right ? 
strange that it accepts null btw